Extensions/TileMap/tilemapruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,146 @@
+
+gdjs.TileMapRuntimeObjectPixiRenderer.prototype.getRendererObject = function() {
+  return this._pixiObject;
+};
+
+gdjs.TileMapRuntimeObjectPixiRenderer.prototype.incrementAnimationFrameX = function(runtimeScene){
+  this._pixiObject.tileAnim[0] += 1;
+};
+
+gdjs.TileMapRuntimeObjectPixiRenderer.prototype.updateTileMap = function(runtimeScene) {
+
+  var atlasTexture = runtimeScene
+    .getGame()
+    .getImageManager()
+    .getPIXITexture(this._object._tilemapAtlasImage);
+  
+  runtimeScene
+    .getGame()
+    .getJsonManager()
+    .loadJson(this._object._tiledFile, (error, content)=> {
+      if (error) {
+        console.error('An error happened while loading JSON resource:', error);
+      } else {
+        if (!content) return;
+        console.log("JSON", content, atlasTexture)
+
+        console.log(
+          this._pixiObject,
+          PixiTileMapHelper,
+          this,
+          this._object._tiledFile,
+          this._object._tilemapAtlasImage,
+          this._object._displayMode
+          )
+  
+          
+        // ReloadTilemap
+        PixiTileMapHelper.getPIXITileSet(
+          atlasTexture,
+          content,
+          this._object._tilemapAtlasImage,
+          this._object._tiledFile,
+          (tileset) => {
+            
+            if (tileset && this._pixiObject) {
+              PixiTileMapHelper.updatePIXITileMap(
+                this._pixiObject,
+                tileset,
+                this._object._displayMode,
+                this._object._layerIndex
+              );
+              console.log('LOADED', tileset, this._pixiObject);
+            }
+          }
+        )
+        // runtimeScene._renderer._pixiRenderer
+        // this._runtimeScene.getTimeManager().getTimeFromStart()
+      }
+    });   
+};
+
+gdjs.TileMapRuntimeObjectPixiRenderer.prototype.updateAnimationFrame = function() {
+  console.log("UPDATING....")
+};
+gdjs.TileMapRuntimeObjectPixiRenderer.prototype.updateTiledFile = function() {
+  this._pixiObject._tiledFile = this._object._tiledFile;
+  //this._pixiObject.dirty = true;
+  this.updatePosition();
+};
+gdjs.TileMapRuntimeObjectPixiRenderer.prototype.updateTilemapAtlasImage = function() {
+  this._pixiObject._tilemapAtlasImage = this._object._tilemapAtlasImage;
+  //this._pixiObject.dirty = true;
+  this.updatePosition();
+};
+gdjs.TileMapRuntimeObjectPixiRenderer.prototype.updateDisplayMode = function() {
+  this._pixiObject._displayMode = this._object._displayMode;
+  //this._pixiObject.dirty = true;
+  this.updatePosition();
+};
+gdjs.TileMapRuntimeObjectPixiRenderer.prototype.updateLayerIndex = function() {
+  this._pixiObject._layerIndex = this._object._layerIndex;
+  //this._pixiObject.dirty = true;
+  this.updatePosition();
+};
+gdjs.TileMapRuntimeObjectPixiRenderer.prototype.updateAnimationSpeed = function() {
+  this._pixiObject._animationSpeed = this._object._animationSpeed;
+  //this._pixiObject.dirty = true;
+  this.updatePosition();
+};
```

Comment:
In all these methods, some properties are set (`_animationSpeed`, `_layerIndex`, `_displayMode`...) but none seems part of `CompositeRectTileLayer`: https://github.com/pixijs/pixi-tilemap/blob/master/src/CompositeRectTileLayer.ts.

So it seems these methods are not working/not doing anything?
Really avoid adding properties if it's not needed. This is destroying any optimization made by the JS engine, because they have to add properties dynamically at runtime on some class that they assumed to be not having these properties. It's also very risky because we're messing with internals of pixi tilemap.

In other words, the rule of thumb is: 1) if it's not documented, don't use it, 2) if it starts with an underscore, you're only allowed to use it if it's yours (i.e: in `this`). If it's another object, it's a big no - don't mess with other objects :) 

I know it's seems to be hardly a big deal, but I'm burn out by spending multiple hours debugging the object when we set it a property that erased one used by pixi tilemap.

This is by the way exactly why we're moving to using TypeScript annotations :)

Reply:
These are not ready yet, sorry, I intend to use them as tuntime variables - not something that is internal to pixi-tilemap but as local variables to the extension.
_animationSpeed is already being used - you can see it in action if you change it.
_layerIndex is too, same as display mode - their update functions are not ready.

_animationSpeed  drives how fast the frame get incremented, so that should already work.

Reply:
all of these variables are required for the extension to work and are passed down to pixi-tilemap - but are not a part of pixi-tilemap. Are you suggesting to attach them to something other than this._object/ this._pixiObject?

Reply:
Sure, but you don't need to put local variables on the pixi object and even `_animationSpeed` here is *not* used :) 
See: `this._pixiObject._animationSpeed` is useless. There is already `this._animationSpeed` in the Runtime object (`gdjs.TileMapRuntimeObject`). It's all that you need, because the object will call `incrementAnimationFrameX`.

And if you need for some reason to store something in the renderer (`gdjs.TileMapRuntimeObjectPixiRenderer`), in this case store it in `this` (and set it to a default value in the constructor). Really there is no reason for storing anything on the pixi object - this is dangerous and the whole reason we have a renderer class is to take care of these details without messing with Pixi/Pixi tilemap internal objects.

Reply:
> Are you suggesting to attach them to something other than this._object/ this._pixiObject?

Yep: in:
* `this._object`. 
* If something is specific to the rendering by PIXI and is not something that would make sense to have on the object (like a pixi texture), then you can store it in `this`, `this` meaning here the `gdjs.TileMapRuntimeObjectPixiRenderer`.

Reply:
added it todo. Need to be able to select image resources for set actions in the event sheet before cleaning this up